### PR TITLE
add files for Peddy analysis pipeline

### PR DESCRIPTION
**Pull request by stellaning1120**
Adding files for new Peddy analysis pipeline based on request from: https://github.com/broadinstitute/TAG/issues/1932

We've done some evaluation for familiar relatedness tools in ticket: https://github.com/broadinstitute/TAG/issues/1704

This WDL is aiming to automate this Peddy analysis process and streamline for growing amount of cases in the future.
This WDL takes pedigree info and individual gVCFs and provide within-family relatedness prediction and runs at sample-set level.

### CODE REVIEW

**Comment by yueyaog** [comment]
@stellaning1120 Can you link an example workspace with this WDL?

**Comment by stellaning1120** [comment]
Here: https://app.terra.bio/#workspaces/broadtagteam/Bahrain_MonogenicCohorts_ClinicalInterp_tag_1741/data

And this is a schematic of the workflow:
Screenshot 2024-08-02 at 11 45 26 AM

**Comment by stellaning1120** [comment]
@yueyaog , Can we review this again for the lastest push?
Actually I find it more complicated than I thought to modify the original WDL to achieve the following target from the VA's suggestion:
proceeded even if single sample families found. The majority of change happens in how we filter passing families in task: FilterSingleSampleFamilies  and GroupFamilyGVCFs .

Also, I modify the original plot script to make it clearer. An example plot below:
concordance-check


I think instead of plotting scatter plot for relatedness, it probably makes more sense to plot concordance result as a categorical variable.

File: PeddyAnalysis/PeddyAnalysis.wdl [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+            for gvcf_path in gvcf_paths:
+                gvcf_base_name = os.path.basename(urlparse(gvcf_path).path)
+                filtered_gVCF_path = f"filtered_{gvcf_base_name}"
+                print(f"Starting to filter: {gvcf_path}")
+                try:
+                    # Filter the gVCF file and ensure the ALT column is all <NON_REF>
+                    import subprocess
+
+                    # Filter the gVCF file and ensure the ALT column is all <NON_REF>
+                    subprocess.run(f'bcftools view -i \'ALT=="<NON_REF>"\' {gvcf_path} | bgzip > {filtered_gVCF_path}', shell=True, check=True)
```
Comment (stellaning1120) [inline PeddyAnalysis/PeddyAnalysis.wdl:379]: check if bcftools has option to output bgzipped file.

File: PeddyAnalysis/PeddyAnalysis.wdl [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+
+        CODE
+    >>>
+
+    output {
+        Array[File] family_gvcfs = glob("filtered_*.gvcf.gz")
+    }
+
+    runtime {
+        docker: "us.gcr.io/tag-public/peddy-analysis:v1"
```
Comment (stellaning1120) [inline PeddyAnalysis/PeddyAnalysis.wdl:401]: take docker override.

File: PeddyAnalysis/PeddyAnalysis.wdl [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+        File reference_fasta_index
+        File reference_dict 
+        File interval_list
+        Array[File] family_gvcfs
+        Int memory = 32
+        Int disk_size = ceil(size(reference_fasta, 'GB') +
+                                size(reference_fasta_index, 'GB') +
+                                size(reference_dict, 'GB') +
+                                (length(family_gvcfs) * 5)) + disk_pad
+        Int disk_pad = 32
```
Comment (stellaning1120) [inline PeddyAnalysis/PeddyAnalysis.wdl:421]: set disk_pad to 0 first as start point.

File: PeddyAnalysis/PeddyAnalysis.wdl [hunk 4]
```diff
@@ -0,0 +1,728 @@
...
+}
+
+task GenotypeGVCFs {
+    input {
+        File combined_gvcf
+        File reference_fasta
+        File reference_fasta_index
+        File reference_dict 
+        String output_prefix
+        Int disk_pad = 32
```
Comment (stellaning1120) [inline PeddyAnalysis/PeddyAnalysis.wdl:462]: same as previous disk_pad.

File: PeddyAnalysis/PeddyAnalysis.wdl [hunk 5]
```diff
@@ -0,0 +1,728 @@
...
+
+    output {
+        File binary_bim = "~{family_id}.bim"
+        File binary_fam = "~{family_id}.fam"
+        File binary_bed = "~{family_id}.bed"
+    }
+
+    runtime {
+        docker: "us.gcr.io/tag-public/peddy-analysis:v1"
+        memory: memory + "GB"
```
Comment (stellaning1120) [inline PeddyAnalysis/PeddyAnalysis.wdl:521]: make docker override

File: PeddyAnalysis/PeddyAnalysis.wdl (hunk 1 above)
Comment (stellaning1120) [reply PeddyAnalysis/PeddyAnalysis.wdl]: I used -Oz to output bgzipped file.

**Review by yueyaog — APPROVED** [review]
This is the correct PR to approve

**Pull request merged**